Source/WebCore/platform/graphics/ca/cocoa/PlatformCALayerCocoa.mm

```diff
@@ -1131,24 +1132,23 @@ - (void)setOwner:(WebCore::PlatformCALayer*)owner
 #endif
 #endif
 
-static NSString *layerContentsFormat(bool wantsDeepColor)
-{
-#if HAVE(IOSURFACE_RGB10)
-    if (wantsDeepColor)
-        return kCAContentsFormatRGBA10XR;
-#else
-    UNUSED_PARAM(wantsDeepColor);
-#endif
-
-    return nil;
-}
-
 void PlatformCALayerCocoa::updateContentsFormat()
 {
     if (m_layerType == PlatformCALayer::LayerType::LayerTypeWebLayer || m_layerType == PlatformCALayer::LayerType::LayerTypeTiledBackingTileLayer) {
         BEGIN_BLOCK_OBJC_EXCEPTIONS
-        if (NSString *formatString = layerContentsFormat(wantsDeepColorBackingStore()))
+        auto contentsFormat = this->contentsFormat();
+
+        if (NSString *formatString = contentsFormatToString(contentsFormat))
             [m_layer setContentsFormat:formatString];
+#if HAVE(HDR_SUPPORT)
+        if (contentsFormat == ContentsFormat::RGBA16F) {
```

Comment:
Should we also check for `ContentsFormat::RGBA10`? EDR is supported with that format too, right? The example from: https://developer.apple.com/documentation/metal/hdr_content/using_color_spaces_to_display_hdr_content?language=objc

uses `.wantsExtendedDynamicRangeContent = YES;` with `.pixelFormat = MTLPixelFormatBGR10A2Unorm;`

Reply:
And possibly relatedly: WebGPU allows the use of RGBA16F but without showing its extended dynamic range when displayed! https://www.w3.org/TR/webgpu/#gpucanvastonemappingmode
This tells me that maybe the contents format shouldn't directly force whether E/HDR should be shown? So we may actually need a separate bit for that.

Reply:
`.wantsExtendedDynamicRangeContent = YES` is set for `ContentsFormat::RGBA10` and `ContentsFormat::RGBA16F`.

For WebGPU case we may need to add a new ContentsFormat enum value.